mpas_analysis/shared/climatology/climatology.py

```diff
@@ -480,11 +494,178 @@ def compute_climatology(ds, monthValues, calendar=None):
+    The start and end years of the climatology are taken from ``config``, and
+    are updated in ``config`` if the data set ``ds`` doesn't contain this
+    full range.
+
+    Note: only works with climatologies where the mask (locations of ``NaN``
+    values) doesn't vary with time.
+
+    Parameters
+    ----------
+    ds : ``xarray.Dataset`` or ``xarray.DataArray`` object
+        A data set with a ``Time`` coordinate expressed as days since
+        0001-01-01
+
+    monthValues : int or array-like of ints
+        A single month or an array of months to be averaged together
+
+    config :  instance of MpasAnalysisConfigParser
+        Contains configuration options
+
+    cachePrefix :  str
+        The file prefix (including path) to which the year (or years) will be
+        appended as cache files are stored
+
+    calendar : ``{'gregorian', 'gregorian_noleap'}``
+        The name of one of the calendars supported by MPAS cores, used to
+        determine ``year`` and ``month`` from ``Time`` coordinate
+
+    printProgress: bool, optional
+        Whether progress messages should be printed as the climatology is
+        computed
+
+    Returns
+    -------
+    climatology : object of same type as ``ds``
+        A data set without the ``'Time'`` coordinate containing the mean
+        of ds over all months in monthValues, weighted by the number of days
+        in each month.
+
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last Modified
+    -------------
+    04/08/2017
+    '''
+    startYearClimo = config.getint('climatology', 'startYear')
+    endYearClimo = config.getint('climatology', 'endYear')
+    yearsPerCacheFile = config.getint('climatology', 'yearsPerCacheFile')
+
+    cacheInfo = []
+
+    ds = add_years_months_days_in_month(ds, calendar)
+
+    if printProgress:
+        print '   Computing and caching climatologies covering {}-year ' \
+              'spans...'.format(yearsPerCacheFile)
+
+    cacheIndices = numpy.zeros(ds.dims['Time'], int)
+    yearsInDs = ds.year.values
+
+    # figure out which files to load and which years go in each file
+    for firstYear in range(startYearClimo, endYearClimo+1, yearsPerCacheFile):
+        years = range(firstYear, numpy.minimum(endYearClimo+1,
+                                               firstYear+yearsPerCacheFile))
+        if len(years) == 0:
+            continue
+
+        if yearsPerCacheFile == 1:
+            yearString = '{:04d}'.format(years[0])
+            outputFileClimo = '{}_year{}.nc'.format(cachePrefix, yearString)
+        else:
+            yearString = '{:04d}-{:04d}'.format(years[0], years[-1])
+            outputFileClimo = '{}_years{}.nc'.format(cachePrefix, yearString)
+
+        done = False
+        if os.path.exists(outputFileClimo):
+            # already cached
+            dsCached = xr.open_dataset(outputFileClimo)
+            if dsCached.attrs['totalMonths'] == 12*len(years):
```

Comment:
Not a big deal but maybe use a constant, `monthsInYear=12` instead of hard-coded 12?